Scripts/RemediationScripts/Remediate-EnableSecurityScanningForContainerRegistry.ps1

```diff
@@ -0,0 +1,716 @@
+        
+        .PARAMETER SubscriptionId
+        Specifies the ID of the Subscription to be remediated.
+        
+        .PARAMETER Force
+        Specifies a forceful remediation without any prompts.
+        
+        .Parameter PerformPreReqCheck
+        Specifies validation of prerequisites for the command.
+        
+        .PARAMETER DryRun
+        Specifies a dry run of the actual remediation.
+        
+        .PARAMETER FilePath
+        Specifies the path to the file to be used as input for the remediation.
+
+        .INPUTS
+        None. You cannot pipe objects to Enable-SecurityScanningIdentityForContainerRegistry.
+
+        .OUTPUTS
+        None. Enable-SecurityScanningIdentityForContainerRegistry does not return anything that can be piped and used as an input to another command.
+
+        .EXAMPLE
+        PS> Enable-SecurityScanningIdentityForContainerRegistry -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -ObjectId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -DryRun
+
+        .EXAMPLE
+        PS> Enable-SecurityScanningIdentityForContainerRegistry -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -ObjectId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck
+
+        .EXAMPLE
+        PS> Enable-SecurityScanningIdentityForContainerRegistry -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -ObjectId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202109131040\EnableSecurityScanningIdentityForContainerRegistry\ContainerRegistryWithoutSecurityScanningEnabled.csv
+
+        .LINK
+        None
+    #>
+
+    param (
+        [String]
+        [Parameter(ParameterSetName = "DryRun", Mandatory = $true, HelpMessage="Specifies the ID of the Subscription to be remediated")]
+        [Parameter(ParameterSetName = "WetRun", Mandatory = $true, HelpMessage="Specifies the ID of the Subscription to be remediated")]
+        $SubscriptionId,
+
+        [Switch]
+        [Parameter(ParameterSetName = "WetRun", HelpMessage="Specifies a forceful remediation without any prompts")]
+        $Force,
+
+        [Switch]
+        [Parameter(ParameterSetName = "DryRun", HelpMessage="Specifies validation of prerequisites for the command")]
+        [Parameter(ParameterSetName = "WetRun", HelpMessage="Specifies validation of prerequisites for the command")]
+        $PerformPreReqCheck,
+
+        [Switch]
+        [Parameter(ParameterSetName = "DryRun", Mandatory = $true, HelpMessage="Specifies a dry run of the actual remediation")]
+        $DryRun,
+
+        [String]
+        [Parameter(ParameterSetName = "DryRun", Mandatory = $true, HelpMessage="Specifies the ID of the object(user) to be assigned as a reader role")]
+        [Parameter(ParameterSetName = "WetRun", Mandatory = $true, HelpMessage="Specifies the ID of the object(user) to be assigned as a reader role")]
```

Comment:
Specifies the ID of the object(user)  ==> Specifies the Object Id of the AAD Identity